test/test_framers.py

```diff
@@ -64,47 +64,78 @@ def test_decode_data(rtu_framer, data):
     assert decoded == expected
 
 
-@pytest.mark.parametrize("data", [(b'', False),
-                                  (b'\x02\x01\x01\x00Q\xcc', True)])
+@pytest.mark.parametrize("data", [
+    (b'', False),
+    (b'\x02\x01\x01\x00Q\xcc', True),
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD', True),  # valid frame
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAC', False),  # invalid frame CRC
+])
 def test_check_frame(rtu_framer, data):
     data, expected = data
     rtu_framer._buffer = data
     assert expected == rtu_framer.checkFrame()
 
 
-@pytest.mark.parametrize("data", [b'', b'abcd'])
-def test_advance_framer(rtu_framer, data):
-    rtu_framer._buffer = data
+@pytest.mark.parametrize("data", [
+    (b'', {'uid': 0x00, 'len': 0, 'crc': b'\x00\x00'}, b''),
+    (b'abcd', {'uid': 0x00, 'len': 2, 'crc': b'\x00\x00'}, b'cd'),
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD\x12\x03',  # real case, frame size is 11
+     {'uid': 0x00, 'len': 11, 'crc': b'\x00\x00'}, b'\x12\x03'),
+])
+def test_rtu_advance_framer(rtu_framer, data):
+    before_buf, before_header, after_buf = data
+
+    rtu_framer._buffer = before_buf
+    rtu_framer._header = before_header
     rtu_framer.advanceFrame()
-    assert rtu_framer._header == {}
-    assert rtu_framer._buffer == data
+    assert rtu_framer._header == {'uid': 0x00, 'len': 0, 'crc': b'\x00\x00'}
+    assert rtu_framer._buffer == after_buf
 
 
 @pytest.mark.parametrize("data", [b'', b'abcd'])
-def test_reset_framer(rtu_framer, data):
+def test_rtu_reset_framer(rtu_framer, data):
     rtu_framer._buffer = data
     rtu_framer.resetFrame()
-    assert rtu_framer._header == {}
+    assert rtu_framer._header == {'uid': 0x00, 'len': 0, 'crc': b'\x00\x00'}
     assert rtu_framer._buffer == b''
 
 
 @pytest.mark.parametrize("data", [
     (b'', False),
+    (b'\x11', False),
+    (b'\x11\x03', False),
     (b'\x11\x03\x06', False),
     (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49', False),
     (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD', True),
-    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD\xAB\xCD', True)
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD\xAB\xCD', True),
 ])
 def test_is_frame_ready(rtu_framer, data):
     data, expected = data
     rtu_framer._buffer = data
-    rtu_framer.advanceFrame()
+    # rtu_framer.advanceFrame()
```

Comment:
This function is useless here and only adds some complexity to the code. It is removed, and some additional test cases are added to be sure that this function works in other cases